lib/linter/linter.js

```diff
@@ -566,6 +633,30 @@ function resolveParserOptions(parser, providedOptions, enabledEnvironments) {
     return mergedParserOptions;
 }
 
+/**
+ * Converts parserOptions to languageOptions for backwards compatibility with eslintrc.
+ * @param {ConfigData} config Config object.
+ * @param {Object} config.globals Global variable definitions.
+ * @param {Parser} config.parser The parser to use.
+ * @param {ParserOptions} config.parserOptions The parserOptions to use.
+ * @returns {LanguageOptions} The languageOptions equivalent.
+ */
+function createLanguageOptions({ globals: configuredGlobals, parser, parserOptions }) {
+
+    const {
+        ecmaVersion,
+        sourceType
+    } = parserOptions;
+
+    return {
+        globals: configuredGlobals,
+        ecmaVersion: normalizeEcmaVersionForLanguageOptions(ecmaVersion),
```

Comment:
If `parserOptions.ecmaVersion` wasn't specified in the configuration, this will set `languageOptions.ecmaVersion` to `NaN`.

If `parserOptions.ecmaVersion` is a string, for example `"next"`, this will set `languageOptions.ecmaVersion` to `"next2019"`.

Reply:
Wow, amazing catch!